charts/ingress-nginx/templates/controller-role.yaml

```diff
@@ -44,12 +44,16 @@ rules:
       - get
       - list
       - watch
+  {{- $updateStatus := default "false" (index .Values.controller.extraArgs "update-status") -}}
+  {{- if eq $updateStatus "true" }}
```

Comment:
Could you please add more information about why we need this?

Reply:
Do you mean in the code?

Reply:
yes. because I think `ingresses/status`  is namespace scored.

Reply:
I get your point, I didn't take a deeper look into the controller's code, yet, and on a green field this will probably work.

But I'm not sure if this is a good idea when disabling the flag on existing deployments, because the current load balancer will definitely remain in the Ingress resources' status field then.

I hopefully find the time to have a deeper look into this soon.

Reply:
Ok, checked it, especially because there's another PR open handling some problems with shutting down controller pods and Ingress resource status:

https://github.com/kubernetes/ingress-nginx/pull/10686
https://github.com/kubernetes/ingress-nginx/issues/10680
https://github.com/kubernetes/ingress-nginx/issues/10681

There's a bit of logic in terminating controller pods which clears the Ingress status if the last Ingress NGINX Controller pod is about to shut down.

Since setting `update-status=false` implies a rollout of the Deployment / DaemonSet due to pod template changes, one will probably end up with Ingresses with outdated / orphaned Ingress resources anyway, as the new pods do not update the status even though they could and the old pods do not clear it as they are told to not do update the status.

I will verify this behavior. Could you please rebase your PR and consider our suggestions in the meantime?

Reply:
Yes sure, I'll do it asap!

Reply:
@Gacko do u suggest removing this?